File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/48-RotateImage/48-RotateImage.cpp
// Last updated: 8/4/2026, 7:24:39 PM
class Solution {
public:
    
    void rotate(vector<vector<int>>& matrix) {
        int n = matrix.size();

        for(int i = 0 ; i < n ; i++){
            for(int j = 0 ; j < n ; j++){
                if(i > j){
                    swap(matrix[i][j] , matrix[j][i]);
                }
            }
        }

        for(int i = 0 ; i < n ; i++){
            reverse(matrix[i].begin() , matrix[i].end());
        }
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/48-RotateImage/48-RotateImage.java
// Last updated: 8/4/2026, 7:24:45 PM
class Solution {
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        int m = matrix[0].length;

        for(int i = 0 ; i < n ; i++){
            for(int j = i ; j < m ; j++){
                if(i != j){
                    int  temp = matrix[i][j];
                    matrix[i][j] = matrix[j][i];
                    matrix[j][i] = temp;
                }
            }
        }

        for(int i = 0 ; i < n ; i++){
            reverse(matrix , 0 , m-1 , i);
        }

    }
    public void reverse(int arr[][] , int i , int j, int row){
        while(i < j){
            int temp = arr[row][j];
            arr[row][j] = arr[row][i];
            arr[row][i] = temp;
            i++;
            j--;

        }
    }
}